src/refer.ts

```diff
@@ -7,33 +7,47 @@ function refer(this: any, options: any) {
     .fix('biz:refer')
     .message('create:entry', actCreateEntry)
     .message('accept:entry', actAcceptEntry)
+    .message('lost:entry', actLostEntry)
+    .message('give:award', actRewardEntry)
     .message('load:rules', actLoadRules)
     .prepare(prepare)
 
   async function actCreateEntry(this: any, msg: any) {
     const seneca = this
-
-    const entry = await seneca.entity('refer/entry').save$({
-      user_id: msg.user_id,
-      kind: msg.kind,
-      email: msg.email,
-
-      // TODO: use a longer key!
-      key: this.util.Nid(), // unique key for this referral, used for validation
-    })
-
-    const occur = await seneca.entity('refer/occur').save$({
-      user_id: msg.user_id,
-      entry_kind: msg.kind,
+    let occur = await seneca.entity('refer/occur').load$({
       email: msg.email,
-      entry_id: entry.id,
-      kind: 'create',
+      kind: 'accept',
     })
+    if (!occur) {
+      const entry = await seneca.entity('refer/entry').save$({
+        user_id: msg.user_id,
+        kind: msg.kind,
+        email: msg.email,
+
+        // TODO: use a longer key!
+        key: this.util.Nid(), // unique key for this referral, used for validation
+      })
+
+      occur = await seneca.entity('refer/occur').save$({
+        user_id: msg.user_id,
+        entry_kind: msg.kind,
+        email: msg.email,
+        entry_id: entry.id,
+        kind: 'create',
+      })
 
+      return {
+        ok: true,
+        entry,
+        occur: [occur],
+      }
+    }
     return {
-      ok: true,
-      entry,
-      occur: [occur],
+      ok: false,
+      why: 'entry-invalid',
```

Comment:
perhaps `entry-exists`? that seems to be the actual issue?

Reply:
Sounds good to me. Done!